### DISCUSSION TITLE
Need help on Cmdline Popup Menu not showing properly

### DISCUSSION THREAD

**Discussion by siraphobk** [folke/noice.nvim#241] [Q&A] [answered]
I enable the command palette in preset and was hoping the output would be similar to the following image as shown in the doc <https://github.com/folke/noice.nvim/wiki/Configuration-Recipes>.

![](https://user-images.githubusercontent.com/292349/193823421-05ca391a-aa31-4ede-b3e0-f27bc74cec7e.png)

However, on my Neovim, the result is

![noice-popup-weird](https://user-images.githubusercontent.com/22819989/203935926-b732906b-9d16-4970-9133-da39b7865e01.png)

Here's my Packer configuration

```lua
 use({
    "folke/noice.nvim",
    config = function()
      require("notify").setup({
        timeout = 1000,
      })

      require("noice").setup({
        lsp = {
          override = {
            ["vim.lsp.util.convert_input_to_markdown_lines"] = true,
            ["vim.lsp.util.stylize_markdown"] = true,
            ["cmp.entry.get_documentation"] = true,
          },
        },
        routes = {
          {
            filter = {
              event = "msg_show",
              kind = "",
              find = "written",
            },
            opts = { skip = true },
          },
        },

        presets = {
          bottom_search = false,
          command_palette = true,
          long_message_to_split = true,
          inc_rename = false,
        }
      })
    end,
    requires = {
      "MunifTanjim/nui.nvim",
      "rcarriga/nvim-notify",
    }
  })
```

And this is the result of healthcheck.

```
noice: require("noice.health").check()

noice.nvim ~
- OK **Neovim** >= 0.8.0
- OK Not running inside **Neovide**
- OK You're not using a GUI
- OK **vim.go.lazyredraw** is not enabled
- OK **nvim-notify** is installed
- OK **treesitter-nvim** is installed
- OK **TreeSitter vim** parser is installed
- OK **TreeSitter regex** parser is installed
- OK **TreeSitter lua** parser is installed
- OK **TreeSitter bash** parser is installed
- OK **TreeSitter markdown** parser is installed
- OK **TreeSitter markdown_inline** parser is installed
- OK `vim.notify` is set to **Noice**
- OK `vim.lsp.handlers["textDocument/hover"]` is set to **Noice**
- OK `vim.lsp.handlers["textDocument/signatureHelp"]` is set to **Noice**
- OK `vim.lsp.handlers["window/showMessage"]` is set to **Noice**
- OK `vim.lsp.util.convert_input_to_markdown_lines` is set to **Noice**
- OK `vim.lsp.util.stylize_markdown` is set to **Noice**
- OK `cmp.entry.get_documentation` is set to **Noice**
```

Neovim version: `NVIM v0.9.0-dev-383+g2738f842f`

---

I believe this might happen because Noice configuration conflicts with some Neovim settings. If you have ideas which setting I should be looking into, please give me your suggestions. Thanks in advance.

**Comment by gegoune** [comment, marked as answer]
It's because of cmp. It displays its popup window. I would like to. know how to solve it too.

**Reply by folke** [reply]
The ony way to make it look like the screenshot is to disable cmp cmdline. You should open an issue/PR at cmp to allow for more styling options of the cmp window. This is not something I'm planning to implement in Noice. It should be fixed in cmp.

**Reply by siraphobk** [reply]
I see. Thanks for your clarification.

**Reply by jrnxf** [reply]
@folke are you suggesting to just remove the `cmp.setup.cmdline({...})` in our dotfiles? I removed all my cmp cmdline related code but now I have no autocompletion when using the cmdline. I'm probably just misunderstanding, but shouldn't noice pick up on this if I have nui as my backend and the popupmenu enabled?

**Reply by folke** [reply]
@thatvegandev if you like the cmp functionality, then just keep it.

The noice popupmenu should work with the default config. It's settings are configured by Neovim, I have little control over that. So it won't automatatically open when you start typing like with cmp. You have to hit `<tab>` or another key if you changed those Neovim options.
